backend/dal/Helpers/Extensions/PropertyExtensions.cs

```diff
@@ -71,6 +71,8 @@ public static IQueryable<Entity.Views.Property> GenerateQuery(this PimsContext c
                 query = query.Where(p => EF.Functions.Like(p.ProjectNumber, $"{filter.ProjectNumber}%"));
             if (filter.IgnorePropertiesInProjects == true)
                 query = query.Where(p => p.ProjectNumber == null);
+            if(filter.InSurplusPropertyProgram == true)
+                query = query.Where(p => !String.IsNullOrWhiteSpace(p.ProjectNumber));
```

Comment:
You will need to test to ensure this query actually sends the correct SQL.  This may actually request all projects from the DB, place them in memory and then check if the `ProjectNumber` value is not null, empty or whitespace.

You can either check **google** to find out if it's a supported function in LINQ EF, or you can output the generated SQL and review.

Reply:
FYI, a while ago I set up a logger such that all of the SQL gets dumped to the output window of VS WHEN DEBUGGING ONLY if you decide to go that route.

Reply:
It is supported! 🎉